internal/app/machined/main.go

```diff
@@ -72,6 +72,37 @@ func sync() {
 	log.Printf("sync hasn't completed in time, aborting...")
 }
 
+func shutdown(seq sequencer.Sequencer, flag int) {
+	if err := seq.Shutdown(); err != nil {
+		panic(fmt.Errorf("shutdown failed: %w", err))
+	}
+
+	sync()
+
+	if unix.Reboot(flag) == nil {
+		select {}
```

Comment:
This is the magic ... every event eventually lands here.

Reply:
one potential way to make it easier to read the code is to put this code after the `for` loop in `main` and break out of the loop when we are ready to reboot/shutdown :) could have used return value of those upgrade/shutdown funcs to decide whether to halt the machine (just an idea)

Reply:
We reboot regardless of success or not. I can look at `break`ing out of the for loop.